### ISSUE TITLE
How to read all data from output shapes of a node?

### ISSUE THREAD

**Issue by PMGruber** [dotnet/machinelearning#5463]
### System information

- **OS version/distro**:
Windows 10
- **.NET Version**: 
.net Core 3.1
- **ML.NET Version**:
ML.NET 1.5.2 
- **Further packages**:
SciSharp.TensorFlow.Redist 2.3.1

### Issue

- **What did you do?**
With my ML.NET code I load an existing TensorFlow Model for object detection and run it. It's an own pretrained model and there is no further training in my code. The execution of the model works fine. The model provides several result information: coordinates, confidences, class ids, number of found objects
- **What happened?**
When I try to read data after inference I only get coordinates.
- **What did you expect?**
Getting all data of node.

### Source code / logs
The special thing about the model is the shape of output node. In all examples I saw so far, it was always e.g. a list of floats. In this specific model the shape are several lists containing the data.

When I have a look at the output node of the model with Netron it looks like this
![image](https://user-images.githubusercontent.com/8576966/97433888-7ce49300-191e-11eb-9500-207614a7130d.png)

My colleagues, consuming the model with python, just call it with something like
`coords, confidence, class_ids,  nums  = model( input_image)`
and receive the results. 

So far I defined my result class as
```C#
public class ImagePrediction
{
    [ColumnName("StatefulPartitionedCall")]
    public float[] Coordinates;
}
```
so it makes sense that only the first list is available as result.

Unfortunately I have no idea what I have do change to get all provided information in ML.NET

Rest of execution code is quite the same like classification sample with a TensorFlow model.

I uploaded model and code to a repository. Since I am not allowed to make the model public, it is a private repo. Please let me know when someone wants to have a deeper look in code/model.

I appreciate any ideas what I can try or even if it is possible/supported at all.

Thanks and regards

**Comment by Lynx1820** [comment]
Hi @PMGruber, 

I took a look, but I'm having some issues. This is what the outermost intermediate node looks like, but I can't see the attributes you described. Can you verify we are looking at the same model? 
![image](https://user-images.githubusercontent.com/10573634/97633068-b4bc0b00-19f0-11eb-8781-bfbc21245476.png)

**Comment by PMGruber** [comment]
Hello @Lynx1820,

thanks for your response.

The node I want to read values from is called "StatefulPartitionedCall". I checked the uploaded model to make sure nothing went wrong during uploading. Netron shows me this
![image](https://user-images.githubusercontent.com/8576966/97691019-17b7ab80-1a9e-11eb-8c6c-a5675922f38a.png)
This is the ouput shape my question refers to.

You are right, Outputs is blank for this node. I get in touch with the creator of the model to ask about it.

**Comment by Lynx1820** [comment]
Hi @PMGruber, please reopen if you have more questions.

**Comment by gionnani** [comment]
Hello.. I have the same problem.

For my model:
(....)
pi = layers.Dense(size, activation='softmax', name='pi')(s_fc4)   
v = layers.Dense(1, activation='tanh', name='v')(s_fc4)               
model = Model(inputs=state, outputs=[pi, v], name="noConv7x4")

The input is ok but the StatefulPartitionedCall (output) shows me only the pi, nothing for v

**Comment by gionnani** [comment]
> Hello.. I have the same problem.
> 
> For my model:
> (....)
> pi = layers.Dense(size, activation='softmax', name='pi')(s_fc4)
> v = layers.Dense(1, activation='tanh', name='v')(s_fc4)
> model = Model(inputs=state, outputs=[pi, v], name="noConv7x4")
> 
> The input is ok but the StatefulPartitionedCall (output) shows me only the pi, nothing for v

I found the way... trying... no place has anything! I've tried put :0 and :1 after StatefulPartitionedCall and it worked!

**Comment by PMGruber** [comment]
Thank you very much @gionnani 
I tried it like you described and it is also working for me. 

What I still not understand is why nodes have different names in ML.NET/Netron and Python...
